### DISCUSSION TITLE
How is "Spatial filer" supposed to work with multiple screen + docking?

### DISCUSSION THREAD

**Discussion by mikkorantalainen** [helloSystem/hello#559] [General]
According to https://github.com/helloSystem/hello/wiki/Welcome-and-unwelcome-technologies#welcome the system has design target "Spatial Filer: Every object in the file system has one place on screen (at defined coordinates), and one place only"

How is this supposed to work when I have a laptop connected to 43" monitor while docked but the built-in screen is obviously much smaller when I'm mobile? How about having 3 monitors while docked and only one small built-in screen when not docked? And note that docking status changes by disconnecting or re-connecting one USB C cable these days, the same cable that's used to attach monitors and charge the laptop battery.

I understand how Spatial Filer used to work in old-school Macintosh computers that only had one screen that was never changed so it made more sense to physically tie objects to pixel positions. However, I fail to see how this is supposed to work in long run with modern environments, even if you specifically decide to not support touch screens at all.

**Comment by probonopd** [comment]
Hi @mikkorantalainen, welcome.

When windows are placed at locations outside of the currently attached screen, then the windows should be moved into the visible area of the attached screen automatically.

Realistically, in the use case you're describing, I'd use the laptop screen as the main screen (for the Filer) and use the other screen(s) as auxiliary screens where I'd move appliction windows to as needed.

(Once could think about saving window locations per screen, but that'd complicate things.)
